docs/e2e-test.md

```diff
@@ -26,3 +26,31 @@ After a finished test, cleanup is performed automatically. If the test setup fai
 ```sh
 make clean -C /opt/metal3-dev-env/metal3-dev-env/ && sudo rm -rf /opt/metal3-dev-env/
 ```
+
+## Included tests
+
+The e2e tests currently run one after the other, using the same environment for all of them.
+They run in this order:
+
+1. Remediation
+1. Inspection *
+1. Pivoting
+1. Upgrade BMO
+1. Upgrade Ironic
+1. Certificate rotation
+1. Node reuse
+1. Re-pivoting
+
+*) Inspection is actually run [in the middle of the remediation test](https://github.com/metal3-io/cluster-api-provider-metal3/blob/8d08f375de93a793f839b42b5ec40e6bebf98664/test/e2e/remediation_test.go#L108) for practical reasons at the moment.
+
+The ephemeral cluster is first launched using [metal3-dev-env](https://github.com/metal3-io/metal3-dev-env).
+The remediation and inspection tests are then run with the controllers still in the ephemeral cluster.
+After this, the controllers are moved to the target cluster during pivoting.
+From this point on all tests are run in the target cluster, until the controllers are moved back to the ephemeral cluster during re-pivoting.
+
+### Upgrade tests
+
+Both Ironic and BMO upgrade tests currently only check that a new version can be rolled out (by going from `latest` to `main`).
+However, they do not actually upgrade from some older release to a newer since we do not yet have the full upgrade test in place for CAPM3.
+Once this is in place, the idea is to upgrade CAPM3 from the previous minor release to the latest.
+Ironic and BMO will then also be upgraded in similar manner using versions that are compatible with CAPM3.
```

Comment:
I would suggest not to merge this as is but rather wait for the proper upgrade to land (consider #466 will be in soon) and then update these notes if there is no real rush on merging it.

Reply:
Sure we can wait with this. #466 is now merged, but I guess it'll be another few weeks until we have refactored the upgrade to include BMO and Ironic. It is fine for me to add the documentation after that but just wanted to explain that it will probably take a while until this part is completed.